File: server/QCLab/Models/Equipment.cs
using System;
using System.Collections.Generic;

namespace QCLab.Models;

public partial class Equipment
{
    public long Id { get; set; }

    public string EquipmentCode { get; set; } = null!;

    public string Name { get; set; } = null!;

    public string? Manufacturer { get; set; }

    public string? Model { get; set; }

    public string SerialNumber { get; set; } = null!;

    public string? Location { get; set; }

    public long StatusId { get; set; }

    public int? CalibrationIntervalDays { get; set; }

    public DateOnly? NextCalibrationDue { get; set; }

    public DateTime DateCreated { get; set; }

    public virtual ICollection<EquipmentCalibration> EquipmentCalibrations { get; set; } = new List<EquipmentCalibration>();

    public virtual EquipmentStatus Status { get; set; } = null!;

    public virtual ICollection<Measurement> Measurements { get; set; } = new List<Measurement>();

    public virtual ICollection<Test> Tests { get; set; } = new List<Test>();
}

File: packages/client_react/src/types/equipment.ts
export interface EquipmentStatusDto {
  id: number;
  name: string;
}

export interface EquipmentCalibrationStatusDto {
  id: number;
  name: string;
}

export interface EquipmentDto {
  id: number;
  equipmentCode: string;
  name: string;
  manufacturer?: string | null;
  model?: string | null;
  serialNumber: string;
  location?: string | null;
  statusId: number;
  status: string;
  calibrationIntervalDays?: number | null;
  nextCalibrationDue?: string | null;
  dateCreated: string;
}

export interface CreateEquipmentDto {
  equipmentCode: string;
  name: string;
  manufacturer?: string | null;
  model?: string | null;
  serialNumber: string;
  location?: string | null;
  statusId: number;
  calibrationIntervalDays?: number | null;
}

export interface UpdateEquipmentDto extends CreateEquipmentDto {}

export interface EquipmentCalibrationDto {
  id: number;
  equipmentId: number;
  calibrationDate: string;
  expirationDate: string;
  certificateNumber: string;
  calibratedBy: string;
  statusId: number;
  resultStatus: string;
  referenceStandardsUsed?: string | null;
  expandedUncertainty?: number | null;
  dateCreated: string;
}

export interface CreateEquipmentCalibrationDto {
  calibrationDate: string;
  expirationDate: string;
  certificateNumber: string;
  calibratedBy: string;
  statusId: number;
  referenceStandardsUsed?: string | null;
  expandedUncertainty?: number | null;
}
